spring-data-jpa/src/test/java/org/springframework/data/jpa/repository/query/HqlQueryTransformerTests.java

```diff
@@ -1043,6 +1047,40 @@ void createsCountQueryUsingAliasCorrectly() {
 		assertCountQuery("select distinct a, count(b) as c from Employee GROUP BY n","select count(distinct a, count(b)) from Employee AS __ GROUP BY n");
 	}
 
+	@Test // GH-3427
+	void sortShouldBeAppendedWithSpacingInCaseOfSetOperator() {
+
+        String source = "SELECT tb FROM Test tb WHERE (tb.type='A') UNION SELECT tb FROM Test tb WHERE (tb.type='B')";
+		String target = createQueryFor(source, Sort.by("Type").ascending());
+
+		assertThat(target).contains(" UNION SELECT ").doesNotContainPattern(Pattern.compile(".*\\SUNION"));
+	}
+
+	@ParameterizedTest // GH-3427
+	@ValueSource(strings = {"", "res"})
+	void sortShouldBeAppendedToSubSelectWithSetOperatorInSubselect(String alias) {
+
+		String prefix = StringUtils.hasText(alias) ? (alias + ".") : "";
```

Comment:
Again I think we should be able to and prefer a simple equals-assertion.

Reply:
I'm not a fan of duplication.